### PR TITLE
Bugfix/points 8 9 11

### PR DESCRIPTION
**Pull request by ValeryTsalko**
ready to test

### CODE REVIEW

File: app/src/main/java/com/example/myweatherapp/adapters/NewsAdapter.kt [hunk 1]
```diff
@@ -11,12 +12,24 @@ import com.example.myweatherapp.models.SourceModel
 import com.example.myweatherapp.viewHolders.NewsHolder
 import com.example.myweatherapp.viewHolders.SourceHolder
 
-open class NewsAdapter(private val newsListener: OnItemClickListener, private val sourceListener: OnSourceClickListener) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+open class NewsAdapter(private val newsListener: OnItemClickListener, private val sourceListener : OnSourceClickListener) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
 
-    var newsList = emptyList<IListItem>()
+    var newsList = mutableListOf<IListItem>()
+    var favoriteUrlList = mutableListOf<String>()
```
Comment (half65) [inline app/src/main/java/com/example/myweatherapp/adapters/NewsAdapter.kt]: Why do you use two lists in an adapter again? It's wrong and you'd better to use only one list here

File: app/src/main/java/com/example/myweatherapp/adapters/NewsAdapter.kt [hunk 2]
```diff
@@ -31,9 +44,9 @@ open class NewsAdapter(private val newsListener: OnItemClickListener, private va
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
         return when (viewType) {
-            0 -> NewsHolder(newsListener, LayoutInflater.from(parent.context).inflate(R.layout.news_holder, parent, false))
-            1 -> SourceHolder(sourceListener, LayoutInflater.from(parent.context).inflate(R.layout.source_holder, parent, false))
-            else -> NewsHolder(newsListener, LayoutInflater.from(parent.context).inflate(R.layout.news_holder, parent, false))
+            0 -> NewsHolder(newsListener, sourceListener, LayoutInflater.from(parent.context).inflate(R.layout.news_holder, parent, false))
```
Comment (half65) [inline app/src/main/java/com/example/myweatherapp/adapters/NewsAdapter.kt]: please make constants for the view types, don't use 1,2,3 here

File: app/src/main/java/com/example/myweatherapp/viewHolders/NewsHolder.kt [hunk 3]
```diff
@@ -6,47 +6,38 @@ import com.bumptech.glide.Glide
...
 
 
-class NewsHolder(private val newsListener: OnItemClickListener, itemView: View) : RecyclerView.ViewHolder(itemView) {
+class NewsHolder(private val newsListener: OnItemClickListener, private val sourceListener: OnSourceClickListener, itemView: View) : RecyclerView.ViewHolder(itemView) {
 
     private val binding = NewsHolderBinding.bind(itemView)
 
-    fun bind(news: NewsModel) = with(binding) {
+    fun bind(news: NewsModel, favoriteUrls: List<String>) = with(binding) {
+        val dateTimeFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy")
```
Comment (half65) [inline app/src/main/java/com/example/myweatherapp/viewHolders/NewsHolder.kt]: please extract it to a global variable. now you're making a new instance again and again in the method

File: app/src/main/res/color/bnm_selector.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+            android:state_checked="true"
+            android:color="@color/selected_BNV"
+            />
+    <item
+            android:color="@color/white" />
+</selector>
```
Comment (half65) [inline app/src/main/res/color/bnm_selector.xml]: EOF please

File: app/src/main/res/drawable/checkbox_selector.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:state_checked="true" android:drawable="@drawable/favorites_icon_filled"/>
+    <item android:drawable="@drawable/favorites_icon_unfilled"/>
+</selector>
```
Comment (half65) [inline app/src/main/res/drawable/checkbox_selector.xml]: the same. EOF

File: app/src/main/res/layout/news_holder.xml [hunk 6]
```diff
@@ -78,14 +80,12 @@
...
-
-        <ImageView
+        <CheckBox
                 android:id="@+id/favorite_icon"
                 android:layout_width="20dp"
                 android:layout_height="20dp"
-                android:layout_marginEnd="15dp"
-                android:layout_marginBottom="10dp"
-                android:background="@drawable/favorites_icon_unfilled"
+                android:layout_marginEnd="20dp"
```
Comment (half65) [inline app/src/main/res/layout/news_holder.xml]: it's a bad practice to put constants directly in the layouts. It's better to extract them into dimens.xml

File: app/src/main/res/layout/spinner_item.xml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+            android:text="text"
+            android:textAlignment="gravity" />
+
+    <CheckBox
+            android:id="@+id/spinner_checkBox"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_alignParentEnd="true"
+            android:layout_alignParentRight="true" />
+</RelativeLayout>
```
Comment (half65) [inline app/src/main/res/layout/spinner_item.xml]: EOF

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -8,8 +8,8 @@
 
     //toolBar
     <string name="tool_bar_news">News</string>
-    <string name="tool_bar_source">Source</string>
-    <string name="tool_bar_favorite">Favorite</string>
+    <string name="tool_bar_source">Sources</string>
+    <string name="tool_bar_favorite">Favorites</string>
 
 
 </resources>
```
Comment (half65) [inline app/src/main/res/values/strings.xml]: EOF

File: app/src/main/java/com/example/myweatherapp/viewHolders/NewsHolder.kt (hunk 3 above)
Comment (ArchiDaKing) [reply app/src/main/java/com/example/myweatherapp/viewHolders/NewsHolder.kt]: dateTimeFormatter was extracted to a global variable

File: app/src/main/res/color/bnm_selector.xml (hunk 4 above)
Comment (ArchiDaKing) [reply app/src/main/res/color/bnm_selector.xml]: fixed

File: app/src/main/res/drawable/checkbox_selector.xml (hunk 5 above)
Comment (ArchiDaKing) [reply app/src/main/res/drawable/checkbox_selector.xml]: fixed

File: app/src/main/res/layout/spinner_item.xml (hunk 7 above)
Comment (ArchiDaKing) [reply app/src/main/res/layout/spinner_item.xml]: fixed

File: app/src/main/res/values/strings.xml (hunk 8 above)
Comment (ArchiDaKing) [reply app/src/main/res/values/strings.xml]: fixed

File: app/src/main/java/com/example/myweatherapp/adapters/NewsAdapter.kt (hunk 2 above)
Comment (ArchiDaKing) [reply app/src/main/java/com/example/myweatherapp/adapters/NewsAdapter.kt]: added companion object with 2 consts

File: app/src/main/res/layout/news_holder.xml (hunk 6 above)
Comment (ValeryTsalko) [reply app/src/main/res/layout/news_holder.xml]: gonna fix it as part of a refactor

File: app/src/main/java/com/example/myweatherapp/adapters/NewsAdapter.kt (hunk 1 above)
Comment (ValeryTsalko) [reply app/src/main/java/com/example/myweatherapp/adapters/NewsAdapter.kt]: left only one list in the adapter

**Comment by half65** [comment]
LGTM

**Pull request merged**